src/pudl/transform/eia.py

```diff
@@ -1100,42 +1100,51 @@ def fix_balancing_authority_codes_with_state(
     ba_name_to_code_map = map_balancing_authority_names_to_codes(plants)
     ba_name_to_code_map.reset_index(inplace=True)
 
-    plants = plants.merge(
-        plants_entity[["plant_id_eia", "state"]],  # only merge in state, drop later
-        on=["plant_id_eia"],
-        how="left",
-        validate="m:1",
-    )
-    BACodeFix = namedtuple(
-        "BACodeFix", ["ba_code_found", "ba_code_fix", "ba_name_fix", "states"]
-    )
-    fixes = [
-        BACodeFix(
-            "PACE",
-            "PACW",
-            ba_name_to_code_map.loc[
-                ba_name_to_code_map.balancing_authority_code_eia == "PACW",
-                "balancing_authority_name_eia",
-            ].tolist()[0],
-            ["OR", "CA"],
-        ),
-        BACodeFix(
-            "PACW",
-            "PACE",
-            ba_name_to_code_map.loc[
-                ba_name_to_code_map.balancing_authority_code_eia == "PACE",
-                "balancing_authority_name_eia",
-            ].tolist()[0],
-            ["UT"],
-        ),
-    ]
-    for fix in fixes:
-        plants.loc[
-            (plants.balancing_authority_code_eia == fix.ba_code_found)
-            & (plants.state.isin(fix.states)),
-            ["balancing_authority_code_eia", "balancing_authority_name_eia"],
-        ] = [fix.ba_code_fix, fix.ba_name_fix]
-    return plants.drop(columns=["state"])
+    logger.info("Spot fixing incorrect PACW/PACE BA codes and names.")
```

Comment:
Maybe put this logging message inside the `if` so it's only emitted if we are actually doing the spot fix.

A comment here explaining why there are sometimes no values to fix would be helpful to our future selves.

Just out of curiosity, is there a reason we aren't using the new spot fixer here instead? Does it not work because one of the columns being fixed is also part of the index that's used to identify records which need fixing?

Reply:
Two reasons spot fixer doesn't work here. The state column is getting manually joined and then removed from another table to enable filtering by state, and we'd skip the step of generating the most common plant name for the plant code and just manually assign it.

But in terms of the other suggestions, great, can do.